File: java-serializers/java-serializer-plugin-test/src/main/java/online/codeisfun/plugins/serializers/test/Data2.java
package online.codeisfun.plugins.serializers.test;

import online.codeisfun.plugins.serializers.CIFSerializable;

import java.util.List;
import java.util.Map;

@CIFSerializable
public class Data2 {
    private Integer a2;
    private String b2;
    private List<Integer> c2;
    private Data1 d2;
    private List<Data1> e2;
    private Map<String, Data1> f2;
    private Map<String, OtherClass2> f3;
}

File: java-serializers/java-serializer-plugin-test-kryo-kotlin/src/main/kotlin/online/codeisfun/plugins/serializers/test/Data2.kt
package online.codeisfun.plugins.serializers.test

import online.codeisfun.plugins.serializers.CIFSerializable

@CIFSerializable
data class Data2(
    val a2: Int? = null,
    val b2: String? = null,
    val c2: List<Int>? = null,
    val d2: Data1? = null,
    val e2: List<Data1>? = null,
    val f2: Map<String, Data1>? = null,
    val f3: Map<String, OtherClass2>? = null,
)
